FirebaseMLModelDownloader/Sources/ModelFileManager.swift

```diff
@@ -75,7 +103,12 @@ class ModelFileManager: FileManager {
       }
     }
     do {
-      try FileManager.default.moveItem(at: sourceURL, to: destinationURL)
+      guard let result = try? hasSufficientSpace(size: size), result == true else {
```

Comment:
I wonder if additional disk space is required to move a file? Can it depend on OS version? As for me, I would rather let the file manager to fail and try to translate its error to an error we are interested in. WDYT?

Reply:
I liked this idea too, but I couldn't figure out how to translate whatever error FileManager throws to be about available space (vs something else). Since that's probably the only I/O error that the developer can actually do something about. How could we handle this better? Also @ryanwilson

Reply:
Looks like [this](https://developer.apple.com/documentation/swift/cocoa_design_patterns/handling_cocoa_errors_in_swift) article uses specifically FileManager case as an example. I guess `CocoaError.fileWriteOutOfSpace` is the error you are interested in, though worth double checking.

Reply:
That is an excellent find, thank you! I'll use `CocoaError.fileWriteOutOfSpace` to isolate the space availability error.